### ISSUE TITLE
[clang] -Weverything option produces a warning(-Wshadow-uncaptured-local) when explicitly capturing a local value using std::move()

### ISSUE THREAD

**Issue by Wayonb** [llvm/llvm-project#81307]
This bug is similar to https://github.com/llvm/llvm-project/issues/71976, except my scenario deals with move capture syntax.

This is a standard C++ pattern when moving objects into a lambda.  Or is there something I am missing?

From my repo, the warning only happens when ``-Weverything`` option is used and not when ``-Wshadow-uncaptured-local`` is used.
https://godbolt.org/z/6v1jbx5f1

```cpp
#include <utility>

void foo();

int main() {
    foo();
    return 0;
}

void foo() {
    int* a = new int(1);
  ([a = std::move(a)]() {
    *a = 2;
  })();
}
```

**Comment by shafik** [comment]
CC @cor3ntin

**Comment by cor3ntin** [comment]
Cc @Fznamznon @erichkeane

**Comment by jarzec** [comment]
I think all cases whenever initialization is used get flagged. I believe initializations with `std::move`, `std::forward` and `std::as_const` (anything else?) should not be flagged.
This is true for version 18.1.6.

**Comment by JarekGlobus** [comment]
I have confirmed for that  v17.0.1, 18.1.0 and 19.1.0 and the original example on Godbolt with all the specific shadow flags:
`-Wshadow-field -Wshadow-field-in-constructor -Wshadow-field-in-constructor-modified -Wshadow-ivar -Wshadow-uncaptured-local`
`std::move()` in capture is not flagged, while if `-Wshadow` or `-Wshadow-all` is added to the above set of flags the code does not compile anymore.

**Comment by jarzec** [comment]
I have confirmed for that  v17.0.1, 18.1.0 and 19.1.0 and the original example on Godbolt with all the specific shadow flags:
`-Wshadow-field -Wshadow-field-in-constructor -Wshadow-field-in-constructor-modified -Wshadow-ivar -Wshadow-uncaptured-local`
`std::move()` in capture is not flagged, while if `-Wshadow` or `-Wshadow-all` is added to the above set of flags the code does not compile anymore.
